itou/www/dashboard/views.py

```diff
@@ -218,8 +219,11 @@ def edit_user_email(request, template_name="dashboard/edit_user_email.html"):
         request.user.email = form.cleaned_data["email"]
         request.user.save()
         request.user.emailaddress_set.all().delete()
+
+        # NOTE: Until the email address is verified, the user will not be able to login
+        send_email_confirmation(request, request.user)
         auth.logout(request)
```

Comment:
Je pense qu'il faudrait mieux ne pas changer l'email pour éviter de bloquer l'utilisateur :


```suggestion
        new_email = form.cleaned_data["email"]
        request.user.emailaddress_set.add(email=new_email)
        send_email_confirmation(request, request.user, email=new_email)
```

Et ensuite tu laisses allauth gérer django-allauth pour changer l'email à la vérification (avec le setting ACCOUNT_CHANGE_EMAIL, voir https://docs.allauth.org/en/dev/account/configuration.html)

Il faudra aussi adapter notre template `email_confirm.html` pour les comptes qui changent leur email afin de ne pas avoir `Ce compte est inactif` quand c'est juste un changement d'email.

Reply:
Je suis d'accord dans l'idéal mais je préfères continuer à bloquer l'utilisateur car il réduit beaucoup la complexité (évite de nécessiter les changements du logique d'unicité des emails). Vu que c'est le comportement actuel et qu'il n'apporte pas un taux au support

J'ai cherché et reçu confirmation d'UX dans les échanges sur la recette jetable

Reply:
Le problème c'est que s'il se trompe d'adresse email, il sera bloqué, non ?

Reply:
Oui exactement

Reply:
@celine-m-s on a parlé de ce sujet mardi

Reply:
Du coup ça va créer du support pour rien, c'est dommage.

Alors qu'avec la solution que je propose, il n'est pas bloqué, et peut demander un autre changement d'adresse.

Reply:
Je viens d'en discuter avec Marion, on attend un retour du support. Dans tous les cas ce sera une solution temporaire, mais la question et si c'est un sujet pour maintenant ou la prochaine trimèstre

Reply:
Je ne comprends pas pourquoi faudrait-il bloquer l'utilisateur.
- Il est connecté. Il entre une nouvelle adresse.
- Il reçoit un email pour vérifier l'adresse. Il oublie de la vérifier.
- Il se connecte avec la nouvelle adresse : renvoi de l'e-mail. Il se connecte avec l'ancienne adresse : ça fonctionne.

Bloquer la connexion des utilisateurs est frustrant alors qu'il est possible de faire autrement.

Reply:
Dans la solution proposée par Antoine, le nouvel e-mail est ajouté mais n'est pas actif tant que l'ancien est l'est. Lors du clic sur le lien, l'ancien e-mail est substitué par le nouveau (l'ancien est désactivé au profit du nouveau qui est activé).

> (évite de nécessiter les changements du logique d'unicité des emails)
Je ne vois pas le lien avec la contrainte d'unicité sur l'e-mail.

Reply:
Je viens de tester et, effectivement, le système bloque l'utilisateur à la modification d'une adresse e-mail. Honnêtement, je ne sais pas pourquoi, mais ce n'est pas intentionnel.
Ta proposition de solution est un premier pas efficace qui règle effectivement un des soucis : l'email ne partait pas au bon moment.
Cela étant, nous pouvons aller plus loin, et c'est ce que propose Antoine.
Au lieu de continuer comme actuellement à bloquer l'utilisateur, nous pouvons le laisser se connecter avec l'adresse active et vérifiée, même si une autre existe.
L'autre avantage que je vois est que cela permet de conserver une trace des changements d'e-mail. Actuellement, on écrase la précédente, ce qui n'est pas génial.
Qu'en penses-tu ? Maintenant que j'ai bien le contexte, nous pouvons en discuter si tu préfères.

Reply:
Je pense que la modification en mode bloquant comporte un certain valeur. Pour l'instant, nous ne prêtons pas beaucoup d'attention à `EmailAddress` et ça simplifie des choses

Cela vaut donc l'effort pour empêcher qu'un utilisateur soit bloqué